remoc/src/codec/postcard.rs

```diff
@@ -0,0 +1,38 @@
+use serde::{Deserialize, Serialize};
+
+use super::{Codec, DeserializationError, SerializationError};
+
+
+
+/// Postcard codec.
+///
+/// See [postcard] for details.
+/// This uses the default function configuration.
+#[cfg_attr(docsrs, doc(cfg(feature = "codec-postcard")))]
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Postcard;
+
+impl Codec for Postcard {
+    #[inline]
+    fn serialize<Writer, Item>(writer: Writer, item: &Item) -> Result<(), super::SerializationError>
+    where
+        Writer: std::io::Write,
+        Item: serde::Serialize,
+    {
+        postcard::to_io(item, writer).map_err(SerializationError::new)?;
+        Ok(())
+    }
+
+    #[inline]
+    fn deserialize<Reader, Item>(mut reader: Reader) -> Result<Item, super::DeserializationError>
+    where
+        Reader: std::io::Read,
+        Item: serde::de::DeserializeOwned,
+    {
+        let mut bytes = vec![];
+        reader.read_to_end(&mut bytes).map_err(DeserializationError::new)?;
+        
+        let item = postcard::from_bytes(&bytes).map_err(DeserializationError::new)?;
```

Comment:
Would it be possible to use postcard::from_io here instead?

We want to avoid the keep the whole incoming data in a memory buffer, if possible.

Reply:
I tried that initially but it required us to pass in a sized slice that is big enough to fit all the data to be deserialised. I could have done something like 

```
let mut bytes = [0; 2048];

postcard::from_io((reader, bytes))...
```

But I figured that would cause more problems than it's worth.

https://docs.rs/postcard/latest/postcard/fn.from_io.html

Reply:
A bit unfortunate, but probably the best we can do for now.

Reply:
Yeah, it's kind of annoying. Maybe I can circle back if postcard makes any further amendments to their API to address this.

I'll raise an issue with them about it.